### PR TITLE
Update CameraConfig To Use Dependency Injection

### LINKED ISSUE #1138: Dependency Inject CameraConfig parameters
### Description of the task

See #960 for more details

The `Camera Config` parameters should be changed to use dependency injection, by passing in the `Camera Config` objects to places that need it. (tip: `grep` or search for usages of `CameraConfig` to find where the data config needs to go). Replace and remove the singleton usages of the parameters (they look like `Util::DynamicParameters::etc`)

**ALSO** please delete the `particle_filter` parameters, and the `particle_filter` folder under the `backend/input` folder. This code is no longer used and should be safe to delete.

### Acceptance criteria

- [x] Inject the CameraControl parameters into all places needed
- [ ] Delete the particle_filter parameters and particle_filter folder

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description

Inject CameraConfig as parameters to the constructor of NetworkClient, as well as changing other files affected by it (namely radio_backend and grsim_backend). Also removed the particle_filter folder and .yaml file.

### Testing Done

Currently builds successfully. There are no tests associated with the files.

### Resolved Issues

Resolves #1138 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/backend/input/network/networking/network_client.cpp [hunk 1]
```diff
@@ -24,6 +26,8 @@ NetworkClient::NetworkClient(std::string vision_multicast_address,
                               gamecontroller_multicast_port);
 
     startIoServiceThreadInBackground();
+
+    sharedConfig = config;
```
Comment (garethellis0) [inline src/software/backend/input/network/networking/network_client.cpp]: This should be in the initializer list for this class, ie:
```
    initial_packet_count(0),
    received_world_callback(received_world_callback),
   sharedConfig(config)
```

https://stackoverflow.com/questions/9903248/initializing-fields-in-constructor-initializer-list-vs-constructor-body

Comment (garethellis0) [reply src/software/backend/input/network/networking/network_client.cpp]: Also `sharedConfig` -> `shared_config` as per our naming convention, but could probably just be `config` to be consistent with other usages of configs in the codebase.

**Review by garethellis0 — CHANGES_REQUESTED** [review]
No major comments, lemme know if you need help with any of the build stuff!

File: src/software/backend/input/network/networking/network_client.cpp (hunk 1 above)
Comment (kelvinkoon) [reply src/software/backend/input/network/networking/network_client.cpp]: Now using initializer list and renamed to use config.

**Review by MathewMacDougall — COMMENTED** [review]
Changes look good! Only thing I don't see is the higher-level code that calls / uses the `network_client` actually passing the config in? Or am I just not seeing something?

**Comment by kelvinkoon** [comment]
> Changes look good! Only thing I don't see is the higher-level code that calls / uses the `network_client` actually passing the config in? Or am I just not seeing something?

Can't believe I missed those, my bad :man_facepalming: 
Made changes to pass in for code using network_client. I haven't had the chance to build because my computer seems to be freezing trying to build... So I'll touch up on it tomorrow morning but otherwise, feel free to review.

**Review by amrm3lm — COMMENTED** [review]
looks good, just needs to build

**Comment by MathewMacDougall** [comment]
Odd that it built before without being given the parameter :thinking:

**Review by MathewMacDougall — DISMISSED** [review]
LGTM :+1:

**Comment by kelvinkoon** [comment]
> Odd that it built before without being given the parameter thinking

Can confirm that both radio_backend and grsim_backend build successfully. Now just awaiting reviews :)

**Review by LiCody — DISMISSED** [review]
LGTM config will probably need to be renamed to camera_config but i can change that in my PR

**Review by amrm3lm — DISMISSED** [review]

File: src/software/backend/input/network/networking/network_client.cpp (hunk 1 above)
Comment (kelvinkoon) [reply src/software/backend/input/network/networking/network_client.cpp]: @garethellis0 can you kindly approve the change? I think this comment is the last thing blocking before I merge so let me know if there are any other problems. Thanks!

**Comment by kelvinkoon** [comment]
> LGTM config will probably need to be renamed to camera_config but i can change that in my PR

@garethellis0 Should I change my config to camera_config instead? Since @LiCody has config changes similar to mine in the same file (i.e network_client.cpp), it seems reasonable to do so.

**Comment by garethellis0** [comment]
> > LGTM config will probably need to be renamed to camera_config but i can change that in my PR
> 
> @garethellis0 Should I change my config to camera_config instead? Since @LiCody has config changes similar to mine in the same file (i.e network_client.cpp), it seems reasonable to do so.

Yep, that makes sense, I'd change it.

**Review by MathewMacDougall — DISMISSED** [review]
:+1: :fire: :shipit:

**Review by garethellis0 — DISMISSED** [review]

**Review by garethellis0 — APPROVED** [review]

**Review by MathewMacDougall — APPROVED** [review]

**Pull request merged**